pkg/target/cluster.go

```diff
@@ -47,6 +57,16 @@ func (cl *Cluster) resolveHosts(path string) ([]*Host, error) {
 		return []*Host{
 			cl.Hosts[jumpHash(path, len(cl.Hosts))],
 		}, nil
+	case conf.LB:
+		cl.Hm.Lock()
+		defer cl.Hm.Unlock()
+		availableHostCount := len(cl.AvailableHosts)
+		if availableHostCount == 0 {
+			return nil, fmt.Errorf("no available hosts left")
+		}
+		return []*Host{
+			cl.AvailableHosts[int(xxhash.Sum64([]byte(path)))%availableHostCount],
```

Comment:
`[]byte(path)` conversion is very expensive, see https://stackoverflow.com/a/14373714 
Please consider switching to *jump* hash.

Reply:
indeed, the byte conversion is expensive. Switched to Sum64String which avoids this, and faster than fnv1.

Reply:
https://github.com/cespare/xxhash/blob/3b82fb7d186719faeedd0c2864f868c74fbf79a1/xxhash_safe.go#L9

Reply:
We can leave this for now. But this may defeat the purpose to search for a faster hash function. Just FYI, the main optimization I did to make Nanotube faster so far was exactly removing this type of conversions. I got x2-3 boost in speed from that.

This just comes to increasing complexity with adding another hash function while not being sure if we will get anything our of it. We need to benchmark more.